radicle-httpd/src/api/json.rs

```diff
@@ -64,6 +71,53 @@ pub(crate) fn blob<T: AsRef<[u8]>>(blob: &Blob<T>, path: &str) -> Value {
     response
 }
 
+/// Creates a [`FileStream`] from `blob`, using `tmp_path` as the file.
+#[allow(dead_code)]
```

Comment:
Why is this needed? :thinking:

Reply:
I haven't added a caller using this method, as I'm not familiar with how `httpd` wants to stream big files yet. Only the test case uses this in this PR.

Reply:
you can re-base on https://github.com/radicle-dev/heartwood/pull/331 and show how it works in practice.

Reply:
I've rebased on #331 and changed the target branch to `dave/add-raw-route`.

Reply:
you haven't shown how this will be used in practice, inside [`file_handler`](https://github.com/radicle-dev/heartwood/commit/34b15f9f16d95de3f78e32ca4610280fdf16eb6c#diff-31099b31af3e5bf7f46f6dc4f2db59fdf633d2e65a3a06f48bfedbff35a29c9cR24).

Reply:
I think I did it [here](https://github.com/radicle-dev/heartwood/blob/446299ef832dce74500e3d5d0871e3efcf41f71d/radicle-httpd/src/raw.rs#L42-L43), no?

Reply:
Hm.. this is not how I was envisioning it. As Fintan mentioned:
> It seems a bit weird that you have to have the bytes in memory to avoid them being in memory -- but perhaps it makes sense.

instead of creating a tmp file, you can write it to a fixed path, e.g. `/tmp/<rid>/<hash>/*path` and inside `file_handler`, first check for existence of the file before loading the blob.

- file exists? stream from disk.
- file doesn't exist? 
  - file greater than `BLOB_STREAM_MIN_BYTES` ? save to disk & then stream
  - file is small ? return directly
  
this way, we only create the file once, and if 5 different people try to download a 200MB binary, server only has allocated 200MB once, for a short amount of time. This however, leaves server open to exhausting disk space. So we need a manager for these caches, a simple FIFO cache with a pre-determined space (e.g. 5 GB) would be ok here.

Reply:
On a second thought, an LRU cache is more appropriate here.

Reply:
In that case, maybe we can go further and read / stream directly from the object file in git tree ?

Reply:
that seems to have its own complications, you can put this PR on hold for now.

Reply:
Is there a way to set the PR state to `hold` officially?

Reply:
> Is there a way to set the PR state to hold officially?

You can turn it back into a draft